From: Wei Yang <richard.weiyang@gmail.com>
Subject: mm/sparse: nr_pages won't be 0
Date: Wed, 19 Jun 2024 01:06:09 +0000

Function subsection_map_init() is only used in free_area_init() in the
loop of for_each_mem_pfn_range().  And we are sure in each iteration of
for_each_mem_pfn_range(), start_pfn < end_pfn.

So nr_pages is not possible to be 0 and we can remove the check.

Link: https://lkml.kernel.org/r/20240619010612.20740-1-richard.weiyang@gmail.com
Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
Cc: Mike Rapoport (IBM) <rppt@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/sparse.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/mm/sparse.c~mm-sparse-nr_pages-wont-be-0
+++ a/mm/sparse.c
@@ -195,9 +195,6 @@ void __init subsection_map_init(unsigned
 	int end_sec_nr = pfn_to_section_nr(pfn + nr_pages - 1);
 	unsigned long nr, start_sec_nr = pfn_to_section_nr(pfn);
 
-	if (!nr_pages)
-		return;
-
 	for (nr = start_sec_nr; nr <= end_sec_nr; nr++) {
 		struct mem_section *ms;
 		unsigned long pfns;
_
